sklearn/ensemble/gradient_boosting.py

```diff
@@ -996,6 +997,17 @@ def fit(self, X, y, sample_weight=None, monitor=None):
         else:
             sample_weight = column_or_1d(sample_weight, warn=True)
 
+        if isinstance(self, GradientBoostingClassifier):
```

Comment:
`_validate_y` already does this label encoding, albeit without a LabelEncoder. It's already specialised to regression/classification. Why not just add sample_weight as a parameter to `_validate_y` and handle it there?

Reply:
That sounds like a neat way of doing this.

And would this checking be done for *any* classification algorithm with such `y` (i.e. in `check_classification_targets`) or just for `GradientBoostingClassifier` (i.e just in `_validate_y`)?

I am not fully aware of how other classification algorithms are supposed respond to a single classification target (i.e whether raise error or not?).